spec/person_spec.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../person'
+
+describe 'Person' do
+  context 'should create an instance of Person class' do
+    person = Person.new(20, 'Jhon', true)
+
+    it 'takes three parameters and returns a Person object' do
+      expect(person.class) == Person
+    end
+    it 'returns correct names of person object instance variables' do
+      expect(person.age).to eq 20
+      expect(person.name).to eq 'Jhon'
+      expect(person.parent_permission).to eq true
+    end
+  end
+
+  context 'should create an instance of Person class' do
+    peter = Person.new(13, 'Peter', false)
+    it 'Correct name' do
+      expect(peter.correct_name).to eq 'Peter'
+    end
```

Comment:
- [ ] Kindly add a test which validates the name of a person for the `correct_name` method in `person.rb`

Reply:
This test has already been written, hence, the project is complete.